src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -862,14 +870,48 @@ public static bool CheckMethodConstraints(
                 method,
                 in args,
                 method.TypeSubstitution,
-                ((MethodSymbol)method.OriginalDefinition).TypeParameters,
+                method.OriginalDefinition.TypeParameters,
                 method.TypeArgumentsWithAnnotations,
                 diagnosticsBuilder,
                 nullabilityDiagnosticsBuilderOpt,
                 ref useSiteDiagnosticsBuilder,
                 skipParameters);
         }
 
+        public static bool CheckMethodConstraintsIncludingExtension(
```

Comment:
>CheckMethodConstraintsIncludingExtension [](http://example.com/codeflow?start=27&length=40)

Why do we want to add a different method instead of simply adjusting behavior of the original method?

Reply:
We discussed offline that we probably want to keep the original name of the method and push the logic that checks constraints on the extension down to the "root" ```CheckConstraints``` helper (so that no one could miss that logic no matter what overload is used).